uyuni-tools/last_update_export.py

```diff
@@ -0,0 +1,103 @@
+4. Customize the directory path, by default `/mnt` was chosen but this could be any location you want, ensure the location has ample free space.
+5. Customize the 'RSYNC_USER' and 'RSYNC_GROUP' in the script to match the user and group names on your system.
+6. Schedule this script using a cron job or another scheduler for daily execution.
+
+Intended Usage:
+- Scheduled daily exports of software channel data.
+- Logging of export operations in '/mnt/logs'.
+- Exports are stored in '/mnt/export/updates'.
+- This script is intended for systems administrators managing software channel updates for Airgapped Environments.
+
+Ensure the server and paths are correctly configured and accessible before running the script.
+"""
+
+import os
+import subprocess
+import datetime
+from xmlrpc.client import ServerProxy
+import ssl
+import socket
+import configparser
+import shutil
+
+# Configuration Variables
+BASE_DIR = "/mnt"  # Define base directory where the exports will be.
+OUTPUT_DIR = os.path.join(BASE_DIR, "export/updates")
+LOG_DIR = os.path.join(BASE_DIR, "logs")
+TODAY = datetime.date.today()
+TARGET_DATE = TODAY - datetime.timedelta(days=1)  # Define the number of days back for export, 1 day by default
+RSYNC_USER = "rsyncuser"  # Define rsync user
+RSYNC_GROUP = "users"     # Define rsync group
+
+def setup_directories(base_path, output_path, log_path):
+    if not os.path.exists(base_path):
+        os.makedirs(base_path, exist_ok=True)
+    if not os.path.exists(log_path):
+        os.makedirs(log_path, exist_ok=True)
+    if os.path.exists(output_path):
+        shutil.rmtree(output_path)
+    os.makedirs(output_path, exist_ok=True)
+
+def setup_logging(LOG_DIR, TODAY):
+    log_file_path = os.path.join(LOG_DIR, f"{TODAY}-daily_export.log")
+    return log_file_path
+
+def create_client():
+    config_path = os.path.expanduser('/root/.mgr-sync')
+    config = configparser.ConfigParser()
+    with open(config_path, 'r') as f:
+        config.read_string('[DEFAULT]\n' + f.read())
+    manager_login = config.get('DEFAULT', 'mgrsync.user')
+    manager_password = config.get('DEFAULT', 'mgrsync.password')
+    suma_fqdn = socket.getfqdn()
+    manager_url = f"https://{suma_fqdn}/rpc/api"
+    context = ssl.create_default_context()
+    client = ServerProxy(manager_url, context=context)
+    return client, client.auth.login(manager_login, manager_password)
+
+# Setup Directories and Logging
+setup_directories(BASE_DIR, OUTPUT_DIR, LOG_DIR)
+log_file_path = setup_logging(LOG_DIR, TODAY)
+
+# Create XML-RPC Client
+client, key = create_client()
+
+# Process channels
+channel_list = client.channel.listVendorChannels(key)
+for channel in channel_list:
+    build_date, channel_label = client.channel.software.getChannelLastBuildById(key, channel["id"]).split()[0], channel["label"]
+    build_date = datetime.datetime.strptime(build_date, "%Y-%m-%d").date()
+
+    if build_date in [TODAY, TARGET_DATE]:
```

Comment:
I think this would only work fine if the interval is 1 day, but since the interval is configurable, it would not behave fine when interval is, i.a. 7 days. The condition would only check if `build_date` is equal to `TODAY` or `TARGET_DATE` but not in between.

I think you could use something like:

```suggestion
    if TARGET_DATE <= build_date <= TODAY:
```